### ISSUE TITLE
Remove old dotfiles and metadata

### ISSUE THREAD

**Issue by floogulinc** [UMM-CSci-Systems/Threading-pre-lab#5]
There are a bunch of old metadata files from InteliJ and Eclipse and such left in the repo. They are mostly called "Pre-lab-7".

`.classpath` in particular causes issues because it specifies JDK 1.6 while some of the stuff in the provided Java sources use features from 1.7+.
https://github.com/UMM-CSci-Systems/Threading-pre-lab/blob/028712f7156c6ebced7e31b519c87c1ab9af6c01/.classpath#L4

You may also consider using Gradle to manage run configs and tests.

**Comment by NicMcPhee** [comment]
I agree that these dotfiles are a mess, but I'm also not sure how to best proceed. The "easy" option is to just replace the Eclipse/IntelliJ files with new VSCode files, but it's annoying to be constantly chasing development environments. I also realize that in a perfect world we wouldn't include _any_ of these, but including them (or at least the right ones) really makes the start of labs a lot simpler since people's code/repos "just works" when they clone it and we're not chasing around trying to figure out how to get the setup right for lots of different people.

Do you have ideas for "better practices" here that would work for us?

**Comment by floogulinc** [comment]
I believe it is recommended to ignore the VS Code automatically created configs. For example, one of the Java ones specify where to find the JDK and what version to use. They are automatically created when VS Code detects a Java project or file. Also note that because VS Code uses Eclipse under the hood, it uses many of the same files like `.classpath`. 

Here is a `.gitignore` created for VS Code and the VS Code Java extension:
```gitignore
# Created by https://www.gitignore.io/api/code,code-java
# Edit at https://www.gitignore.io/?templates=code,code-java

### Code ###
.vscode/*
!.vscode/settings.json
!.vscode/tasks.json
!.vscode/launch.json
!.vscode/extensions.json

### Code-Java ###
# Language Support for Java(TM) by Red Hat extension for Visual Studio Code - https://marketplace.visualstudio.com/items?itemName=redhat.java

.project
.classpath
factoryConfiguration.json

# End of https://www.gitignore.io/api/code,code-java
```

Though I would also suggest ignoring `.settings` and letting it be generated as well, especially if using something like Gradle.
